docs/ref/settings.txt

```diff
@@ -1503,8 +1503,10 @@ exclusive, so only set one of those settings to ``True``.
 Default: ``None``
 
 If :setting:`EMAIL_USE_SSL` or :setting:`EMAIL_USE_TLS` is ``True``, you can
-optionally specify the path to a PEM-formatted certificate chain file to use
-for the SSL connection.
+optionally specify the path to a PEM-formatted client authentication
+certificate chain file to use for the SSL connection.
+It must be used in combination with :setting:`EMAIL_SSL_KEYFILE` or include
+the private key.
```

Comment:
I've been reading https://docs.python.org/3/library/ssl.html#ssl.SSLContext.load_cert_chain which links to https://docs.python.org/3/library/ssl.html#ssl-certificates

I wonder if we should borrow some of the wording here or link to these :thinking: so maybe
```suggestion
optionally specify the path to a certificate chain file, a PEM-formatted file
containing the certificate as well as any number of CA certificates needed to
establish the certificate’s authenticity to use for the SSL connection. It must
be used in combination with :setting:`EMAIL_SSL_KEYFILE` or include the private
key.
```

Reply:
@medmunds unsure if you have thoughts here :pray:

Reply:
I'm not sure how to word it, but I think there are two things we need to convey:

1. EMAIL_SSL_CERTFILE  and EMAIL_SSL_KEYFILE are intended for SMTP *client* authentication, where the client (the Django app trying to send mail) authenticates itself to the SMTP server by proving it has the private key matching the public key in certfile. (And that private key can either be in the keyfile or included in the certfile.)
2. If you are instead trying to work with an SMTP server using a self-signed or private-certificate-authority *server* certificate, these settings don't really allow that. Instead, you should add the *public* key for your root CA or self-signed certificate to your system's trust store so that ordinary TLS to that server works on the machine running Django. (E.g., with openssl you might need to set the SSL_CERT_DIR or SSL_CERT_FILE *environment* variables.) In this case, you should *not* set Django's EMAIL_SSL_{CERT,KEY}FILE settings.

I *suspect* the words "certificate chain file" might be one source of confusion leading people to believe EMAIL_SSL_CERTFILE can be somehow used for the second purpose. (Because I tend to think of cert chaining as something you need to worry about with a private CA.) But I've never come across an SMTP server that used cert based client auth, so maybe there's something I don't understand about cert chaining in scenario 1. (Also, the other source of confusion is that prior to Django 4.1 using EMAIL_SSL_CERTFILE for a private or self-signed cert *appeared* to work—though I guess essentially by disabling cert checking?) 

(Also, #18456 would provide a platform-independent way to address case 2.)

Reply:
How about something like this (replacing the entire description for the setting):

```rst
If :setting:`EMAIL_USE_SSL` or :setting:`EMAIL_USE_TLS` is ``True`` and the
secure connection to your SMTP server requires client authentication, specify
the path to a PEM-formatted client authentication certificate chain file.
It must be used in combination with :setting:`EMAIL_SSL_KEYFILE` or include
the private key.

:setting:`!EMAIL_SSL_CERTFILE` cannot be used for a self-signed server
certificate or a private certificate authority root. In most cases, you should
install that in the system CA bundle. Look for instructions for installing a
root CA certificate on your platform, or for using OpenSSL's SSL_CERT_FILE
or SSL_CERT_DIR environment variables with a custom bundle if you don't want
to modify the system bundle. (For more complex cases, you can subclass the
SMTP EmailBackend and add root certificates to its ``ssl_context`` using
:meth:`python:ssl.SSLContext.load_verify_locations`.)
```

I stuck with @dbIgel's original wording, and added a paragraph covering the unsupported case that I suspect is tripping people up. I don't think it's necessary to go into details about certificate chaining here. (Environments that use *client certificate* SMTP authentication probably have someone familiar with all of that responsible for managing their certs.)